components/director/internal/domain/apptemplate/service.go

```diff
@@ -344,6 +349,18 @@ func (s *service) retrieveLabel(ctx context.Context, id string, labelKey string)
 	return label.Value, nil
 }
 
+func validatePlaceholderValue(placeholder model.ApplicationTemplatePlaceholder, value string) error {
+	if placeholder.Name == "provider" && value == providerSAP {
+		return errors.New("provider cannot be SAP")
+	}
+
+	if placeholder.Name == "application-type" && (strings.HasPrefix(value, providerSAP+" ") || value == providerSAP) {
```

Comment:
Similar to cockpit change. Get first 4 symbols, [trim](https://pkg.go.dev/strings#TrimSpace) the result and check if it is "SAP"